src/QMCDrivers/VMC/VMCFactoryNew.cpp

```diff
@@ -2,47 +2,50 @@
 // This file is distributed under the University of Illinois/NCSA Open Source License.
 // See LICENSE file in top directory for details.
 //
-// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+// Copyright (c) 2019 QMCPACK developers.
 //
-// File developed by: Ken Esler, kpesler@gmail.com, University of Illinois at Urbana-Champaign
+// File developed by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
+//                    Ken Esler, kpesler@gmail.com, University of Illinois at Urbana-Champaign
 //                    Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
 //                    Jeremy McMinnis, jmcminis@gmail.com, University of Illinois at Urbana-Champaign
 //                    Raymond Clay III, j.k.rofling@gmail.com, Lawrence Livermore National Laboratory
 //
 // File created by: Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
 //////////////////////////////////////////////////////////////////////////////////////
 
-
 #include "QMCDrivers/VMC/VMCFactoryNew.h"
 #include "QMCDrivers/VMC/VMCBatched.h"
-#include "Message/OpenMP.h"
-
-#ifdef QMC_CUDA
-#include "QMCDrivers/VMC/VMC_CUDA.h"
-#endif
+//#include "Message/OpenMP.h"
+#include "Concurrency/Info.hpp"
 
 namespace qmcplusplus
 {
 QMCDriverInterface* VMCFactoryNew::create(MCPopulation& pop,
-                                       TrialWaveFunction& psi,
-                                       QMCHamiltonian& h,
-                                       ParticleSetPool& ptclpool,
-                                       HamiltonianPool& hpool,
-                                       WaveFunctionPool& ppool,
-                                       Communicate* comm)
+                                          TrialWaveFunction& psi,
+                                          QMCHamiltonian& h,
+
+                                          ParticleSetPool& ptclpool,
+                                          HamiltonianPool& hpool,
+                                          WaveFunctionPool& ppool,
+                                          Communicate* comm)
 {
-  int np = omp_get_max_threads();
-  //(SPACEWARP_MODE,MULTIPE_MODE,UPDATE_MODE)
+  QMCDriverInput qmcdriver_input(qmc_counter_);
+  qmcdriver_input.readXML(input_node_);
+  VMCDriverInput vmcdriver_input;
+  vmcdriver_input.readXML(input_node_);
   QMCDriverInterface* qmc = nullptr;
-  if (VMCMode == 0 || VMCMode == 1) //(0,0,0) (0,0,1)
+
+  if (vmc_mode_ == 0 || vmc_mode_ == 1) //(0,0,0) (0,0,1)
   {
-    qmc = new VMCBatched(pop, psi, h, ppool, comm);
+    qmc = new VMCBatched(qmcdriver_input, vmcdriver_input, pop, psi, h, ppool, comm);
```

Comment:
I feel better having vmcdriver_input derived from qmcdriver_input following the nature that vmcdriver is derived from qmcdriver.

Reply:
We can continue to discuss this is just a first factoring of the Input out of the driver.